tests/components/mikrotik/test_config_flow.py

```diff
@@ -0,0 +1,208 @@
+        result["flow_id"], user_input=DEMO_USER_INPUT
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == "Home router"
+    assert result["data"][CONF_NAME] == "Home router"
+    assert result["data"][CONF_HOST] == "0.0.0.0"
+    assert result["data"][CONF_USERNAME] == "username"
+    assert result["data"][CONF_PASSWORD] == "password"
+    assert result["data"][CONF_PORT] == 8278
+
+
+async def test_options(hass):
+    """Test updating options."""
+    entry = MockConfigEntry(domain=mikrotik.DOMAIN, data=DEMO_CONFIG_ENTRY)
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.options.async_init(entry.entry_id)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "device_tracker"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={
+            mikrotik.CONF_DETECTION_TIME: 30,
+            mikrotik.CONF_ARP_PING: True,
+            mikrotik.const.CONF_FORCE_DHCP: False,
+        },
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["data"] == {
+        mikrotik.CONF_DETECTION_TIME: 30,
+        mikrotik.CONF_ARP_PING: True,
+        mikrotik.const.CONF_FORCE_DHCP: False,
+    }
+
+
+async def test_host_already_configured(hass, auth_error):
+    """Test host already configured."""
+
+    entry = MockConfigEntry(domain=mikrotik.DOMAIN, data=DEMO_CONFIG_ENTRY)
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        mikrotik.DOMAIN, context={"source": "user"}
+    )
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input=DEMO_USER_INPUT
+    )
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
+
+
+async def test_name_exists(hass, api):
+    """Test name already configured."""
+
+    entry = MockConfigEntry(domain=mikrotik.DOMAIN, data=DEMO_CONFIG_ENTRY)
+    entry.add_to_hass(hass)
+    user_input = DEMO_USER_INPUT.copy()
+    user_input[CONF_HOST] = "0.0.0.1"
+
+    result = await hass.config_entries.flow.async_init(
+        mikrotik.DOMAIN, context={"source": "user"}
+    )
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input=user_input
+    )
+
+    assert result["type"] == "form"
+    assert result["errors"] == {CONF_NAME: "name_exists"}
+
+
+async def test_connection_error(hass, conn_error):
+    """Test error when connection is unsuccesful."""
+
+    result = await hass.config_entries.flow.async_init(
+        mikrotik.DOMAIN, context={"source": "user"}
+    )
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input=DEMO_USER_INPUT
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "cannot_connect"}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4918093)